fiftyone/core/dataset.py

```diff
@@ -136,11 +143,18 @@ def load_dataset(name):
 
     Args:
         name: the name of the dataset
+        create_if_missing (False): if no dataset exists, create empty one
 
     Returns:
         a :class:`Dataset`
     """
-    return Dataset(name, _create=False)
+    try:
+        return Dataset(name, _create=False)
+    except MissingDatasetError as ex:
+        if create_if_missing:
+            return Dataset(name)
+        else:
+            raise ex
```

Comment:
I'd use `fo.dataset_exists()` over try-except.

Reply:
Good idea, done!

Reply:
I disagree with @sashankaryal 's comment. While it might look nicer, this causes an extra database lookup for the happy path.
We should assume it exists and fall back to creating rather than checking for existence.

Reply:
Hmm, how much do we want to optimize database lookup? I don't think the users will be calling `load_dataset` very often, probably once at the beginning of the script.

Reply:
you might be surprised. especially when going through pymongo proxy (dont wanna get into it lol). where each call to mongodb must be absolutely necessary haha.

Reply:
I see, @sashankaryal what do you think?

Reply:
Chatted with @sashankaryal, let's go with @swheaton's suggestion